client/store/modules/comments/mutations.js

```diff
@@ -16,7 +15,6 @@ const commentsFetched = (state, activityId) => {
 const sseAdd = (state, comment) => {
   const { id } = comment;
   if (find(state.items, { id })) return;
-  comment._cid = cuid();
```

Comment:
Here was actually the main problem where we set the cuid without mapping it.
So this happens:

```js
const sseAdd = (state, comment) => {
  const { id } = comment;
  if (find(state.items, { id })) return;
  comment._cid = cuid();
  Vue.set(state.items, comment._cid, comment);
};
```
1. SSE event is triggered and the item has a new CUID which is not mapped just set
2. User selects the repository and **fetch** is triggered which activates **setCuid()**
 ```js
  getCid(id) {
    return this.mappings[id];
  }
setCid(model) {
    model._cid = this.getCid(model.id) || cuid();
    if (model.id) this.map(model._cid, model.id);
  }

```
3. setCuid()  is now searching for the cuid in mapping and as it doesn't find it so it sets it to a new cuid.
4. Now when we are filling the state and expecting that the same comments (with the same cuid) are overwritten this doesn't happen and we have comment duplication. 

``